messaging-client-databus/src/main/java/com/inmobi/messaging/consumer/databus/DatabusConsumer.java

```diff
@@ -181,47 +206,22 @@ protected void createPartitionReaders() throws IOException {
             ((Checkpoint) currentCheckpoint).getPartitionsCheckpoint();
         LOG.info("Creating partition readers for all the collectors");
         for (String collector : getCollectors(fs, streamDir)) {
-          PartitionId id = new PartitionId(clusterName, collector);
-          PartitionCheckpoint pck = partitionsChkPoints.get(id);
-          /*
-           * Migration of checkpoint required in this case
-           * If user provides a cluster name and partition checkpoint is null
-           */
-          if (!clusterName.equals(getDefaultClusterName(i)) && pck == null) {
-            PartitionId defaultPid = new PartitionId(getDefaultClusterName(i),
-                collector);
-            pck = partitionsChkPoints.get(defaultPid);
-            /*
-             * Migrate to new checkpoint
-             */
-            ((Checkpoint) currentCheckpoint).migrateCheckpoint(pck, defaultPid, id);
-          }
-          Date partitionTimestamp = getPartitionTimestamp(id, pck);
-          LOG.debug("Creating partition " + id);
-          PartitionReaderStatsExposer collectorMetrics = new
-              CollectorReaderStatsExposer(topicName, consumerName,
-                  id.toString(), consumerNumber, fsuri);
-          addStatsExposer(collectorMetrics);
-          Path streamsLocalDir = null;
-          if (readFromLocalStream) {
-            streamsLocalDir = DatabusUtil.getStreamDir(StreamType.LOCAL,
-                rootDirs[i], topicName);
-          }
-          for (int c = 0; c < numList; c++) {
-            collectorMetrics.incrementListOps();
+          if (collectorsWithReaders.contains(collector + rootDirs[i])) {
+            continue;
           }
-          readers.put(id, new PartitionReader(id, pck, conf, fs,
-              new Path(streamDir, collector), streamsLocalDir, buffer, topicName,
-              partitionTimestamp, waitTimeForFlush, waitTimeForFileCreate,
-              collectorMetrics, stopTime));
-          messageConsumedMap.put(id, false);
-          numList = 0;
+          String defaultClusterName = getDefaultClusterName(i);
+          Path rootdir = rootDirs[i];
+          createPartitionReader(clusterName, collector, partitionsChkPoints, fsuri, fs, streamDir, defaultClusterName, rootdir);
+          LOG.info("Creating partition reader Collector " + collector);
```

Comment:
Move this log before calling createPartitionReader(). Preferably have another message after, saying "created partition reader for collector " + collector.